pilot/cmd/pilot-agent/main.go

```diff
@@ -256,6 +257,10 @@ var (
 			}
 
 			log.Infof("Monitored certs: %#v", certs)
+			// since Envoy needs the certs for mTLS, we wait for them to become available before starting it
+			if controlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS.String() {
+				waitForCerts(path.Join(certs[0].Directory, certs[0].Files[0]), 2*time.Minute)
```

Comment:
Why only on first element of the slice? Maybe loop through all to make sure all certs are avail?

Reply:
Also, I don't understand what happens if certs aren't available after the max time? It just continues?

Reply:
If certs aren't available, it falls back to the current behavior (letting Envoy try and possibly restart). There isn't that much we can do and continuing forever does not add value.

The assumption is that once one certificate file is ready, so are the others. If this is too lax, I can certainly loop through `certs` array and check first file from each. I believe all certs in a directory are coming from the same secret (i,e., cert, key and chain are all parts of one secret)

Reply:
> If certs aren't available, it falls back to the current behavior (letting Envoy try and possibly restart). There isn't that much we can do and continuing forever does not add value.

Fair enough but maybe print some summarising error message?